src/Providers/ProviderRow.tsx

```diff
@@ -0,0 +1,148 @@
+
+import { ConfirmModal } from '@app/common/components/ConfirmModal';
+import { ProviderType } from '@app/common/constants';
+import { useDeleteProviderMutation } from '@app/queries';
+import { StatusIcon } from '@migtools/lib-ui';
+import { ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Popover,
+} from '@patternfly/react-core';
+import { Td, Tr } from '@patternfly/react-table';
+
+import { MergedProvider } from './data';
+
+interface CellProps {
+  value: string;
+  entity: MergedProvider;
+}
+const StatusCell = ({ value, entity: { conditions } }: CellProps) => {
+  const { t } = useTranslation();
+  const existingConditions = Object.values(conditions).filter(Boolean);
+  const toState = (value) =>
+    value === 'True' ? 'Ok' : value === 'False' ? 'Error' : 'Unknown';
+  return (
+    <Popover
+      hasAutoWidth
+      bodyContent={
+        <div>
+          {existingConditions.length > 0
+            ? existingConditions.map(({ message, status }) => {
+                return (
+                  <StatusIcon
+                    key={message}
+                    status={toState(status)}
+                    label={message}
+                  />
+                );
+              })
+            : 'No information'}
+        </div>
+      }
+    >
+      <Button variant="link" isInline aria-label={t(value)}>
+        <StatusIcon status={toState(value)} label={t(value)} />
+      </Button>
+    </Popover>
+  );
+};
+
+const TextCell = ({ value }: { value: string }) => <>{value}</>;
+
+const ProviderLink = ({ value, entity }: CellProps) => (
+  <ResourceLink kind={entity.kind} name={value} namespace={entity?.namespace} />
+);
+
+const cellCreator = {
+  [NAME]: ProviderLink,
+  [READY]: StatusCell,
+  [URL]: TextCell,
+  [TYPE]: TextCell,
+  [NAMESPACE]: ({ value }: CellProps) => (
+    <ResourceLink kind="Namespace" name={value} />
+  ),
+};
+
+const ProviderRow = ({ columns, entity }: RowProps<MergedProvider>) => {
+  const { t } = useTranslation();
+  const [isActionMenuOpen, setIsActionMenuOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  console.warn('Modal open?', isDeleteModalOpen, entity);
+  const toggleDeleteModal = () => setIsDeleteModalOpen(!isDeleteModalOpen);
+  const deleteProviderMutation = useDeleteProviderMutation(
+    entity.type as ProviderType,
+    toggleDeleteModal,
+  );
+  const editProvider = () => '';
+  const selectNetwork = () => '';
+  const isTarget = (type: ProviderType) => type !== 'openshift';
+  return (
+    <Tr>
+      {columns.map(({ id }) => (
+        <Td key={id} dataLabel="foo">
+          {cellCreator?.[id]?.({
+            value: entity[id],
+            entity,
+          }) ?? <TextCell value={String(entity[id] ?? '')} />}
+        </Td>
+      ))}
+      <Td modifier="fitContent">
+        <Dropdown
```

Comment:
note: it will be nice to have the table actions (  plus modals that need to open ) as a prop provided to the table.

note II: in the console implementation you can provide the actions as a prop, ( method that take the `rowData` and return the `<Dropdown>` ) but they forgot to implement actions that open a modal

Reply:
The console have a way to link a list of actions (a kebab dropdown with actions) that will show when the console show a resource, for example in topology view, default tables and default details pages.
https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/docs/console-extensions.md#consoleactionresource-provider

Reply:
I need to do more research here but the first consequence of having global actions is having global data i.e. resources. However our actions run on `flattened` data tailored specifically for this screen. 
Some solution:
1.  our `flat` entity is a direct subset of the resource (structurally)
2. the hook providing the action needs to flatten the data (and potentially fetch extra data i.e. inventory)

> it will be nice to have the table actions ( plus modals that need to open ) as a prop provided to the table.
>

The actions are placed in the row because logically the row is the entity and actions are methods called on the entity.
Code re-use by passing the actions to the `TableView` would require a generic implementation (`TableView` is generic) which is always less flexible (the `ProviderRow` knows all the details of the `MergedProvider` data type).
However there is no problem with passing modals and actions to the row mapper - both `ProviderRow` and `ProviderPage` are non-generic (top-level).

Reply:
Addressed in commit https://github.com/kubev2v/forklift-console-plugin/pull/33/commits/9cdb7631192340c048e926ea511bcf4faef10cfd